pkg/provisioner/ironic/ironic.go

```diff
@@ -1176,9 +1176,14 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 func (p *ironicProvisioner) changePower(ironicNode *nodes.Node, target nodes.TargetPowerState) (result provisioner.Result, err error) {
 	p.log.Info("changing power state")
 
-	// If we're here, we're going to change the state and we should
-	// wait to try that again.
-	result.RequeueAfter = powerRequeueDelay
+	switch nodes.ProvisionState(ironicNode.ProvisionState) {
+	case nodes.Inspecting, nodes.InspectWait, nodes.InspectFail, nodes.Cleaning, nodes.CleanWait:
+		p.log.Info("host in state that does not allow power change, try again after delay",
+			"state", ironicNode.ProvisionState,
+		)
+		result.Dirty = true
```

Comment:
Should this also be setting `result.RequeueAfter`?

Reply:
Good eye, yes.